### PR TITLE
Fixes 0715

### PR DESCRIPTION
**Pull request by lowriech**
Just a starting point for the meeting.  Already deployed to staging.

### CODE REVIEW

**Comment by lowriech** [comment]
I think I want to do a comprehensive overhaul with you and me on this, in the context of the Data Explorer (Asset Viewer) as well.  Let's put some thought into what revamping the data model looks like here.

File: src/compass/compass.js [hunk 1]
```diff
@@ -22,9 +26,85 @@ function reverseObject(obj) {
...
+}
+
+function MangroveContextLayer({
+  map,
+  mapLoaded,
+  theme,
+  action,
+  selectedLayer,
+}) {
+  console.log(selectedLayer);
```
Comment (gislawill) [inline src/compass/compass.js:40]: Could you remove this console log?

File: src/compass/compass.js [hunk 2]
```diff
@@ -62,29 +142,41 @@ export default function Compass(props) {
...
       <div className="controls-panel" ref={props._ref}>
-        <div
+        <MangroveContextLayer
+          map={props.map}
+          mapLoaded={props.mapLoaded}
+          theme={props.theme}
+          selectedLayer={props.selectedLayer}
+          action={setTooltipText}
+        />
+        {/* <div
```
Comment (gislawill) [inline src/compass/compass.js:152]: Could you delete this commented out code?

File: src/components/hover.tsx [hunk 3]
```diff
@@ -3,6 +3,7 @@ import "./hover.css";
 
 interface HoverProps {
   children: ReactNode;
+  action: Function;
```
Comment (gislawill) [inline src/components/hover.tsx:6]: Love the use of TS! It's typically best practice to be as specific as possible with type definitions, especially for functions. In this context, I think the type we'll want to use is
```suggestion
  action: (newValue: string | null) => void;
```
That tells your function that it needs to pass in either string or null and it shouldn't expect anything to be returned

File: src/components/hover.tsx [hunk 4]
```diff
@@ -31,20 +33,22 @@ const Hover: React.FC<HoverProps> = ({
...
   return (
     <div
       className={`${classlist.current}`}
-      onMouseEnter={() => setShow(true)}
-      onMouseLeave={() => setShow(false)}
+      onMouseEnter={() => {
+        action(text);
+      }}
+      onMouseLeave={() => {
+        action(null);
```
Comment (gislawill) [inline src/components/hover.tsx:48]: I wonder if we need to wrap this in a timeout to allow the css "fade out" animation to run before we remove the content from the dom

File: src/hooks/useBreadcrumbs.js [hunk 5]
```diff
@@ -137,6 +138,7 @@ export function useMapWithBreadcrumbs(
 
   useEffect(() => {
     if (!payload) return;
+    console.log(payload);
```
Comment (gislawill) [inline src/hooks/useBreadcrumbs.js:141]: Could you remove this console log?

File: src/hooks/useBreadcrumbs.js [hunk 6]
```diff
@@ -146,22 +148,22 @@ export function useMapWithBreadcrumbs(
...
+  //   0,
+  // );
+
+  // useWhile.off(
+  //   () => !isHovering,
+  //   [isHovering],
+  //   "FIRST_HOVER",
+  //   undefined,
+  //   HOVER_TIMEOUT,
+  // );
```
Comment (gislawill) [inline src/hooks/useBreadcrumbs.js:166]: Let's either delete this code or leave a comment in the code explaining why we should keep it for the future

File: src/panels/stats-panel/stats-panel-container.js [hunk 7]
```diff
@@ -131,9 +126,11 @@ function TopBanner({
...
       </p>
       {selectedYear != initialYear && (
-        <i className="text-right text-white mb-2">
-          Data in the map shows 2015 values.
-        </i>
+        <>
+          <i className="text-left text-white mb-2 text-lg text-[14px]">
+            Data in the map shows 2015 values.
+          </i>
+        </>
```
Comment (gislawill) [inline src/panels/stats-panel/stats-panel-container.js:133]: I don't think we need a fragment here

File: src/compass/compass.js [hunk 8]
```diff
@@ -22,9 +26,85 @@ function reverseObject(obj) {
 
 const ReversedBasemapMap = reverseObject(BasemapMap);
 
+function Tooltip({ text }) {
+  return <div className="tooltips-slot">{text}</div>;
+}
+
+function MangroveContextLayer({
```
Comment (gislawill) [inline src/compass/compass.js:33]: I think this name implies we're using react context here to manage state but it's (thankfully) simpler than that. Could we rename it something like _MangroveLayerToggle_?

**Comment by gislawill** [comment]
Heads up @lowriech, updated the base branch for this PR to focus the new additions. Also opened a staging => main PR that we can merge once the build succeeds

File: src/compass/compass.js (hunk 8 above)
Comment (gislawill) [reply src/compass/compass.js:33]: Not new code, resolving

File: src/compass/compass.js [hunk 9]
```diff
@@ -60,6 +70,14 @@ function MangroveContextLayer({ map, theme, action }) {
     else addLayer();
   }, [mangrovesOn]);
 
+  useEffect(() => {
+    if (selectedLayerRef.current == LayerName.Flooding) {
+      setMangrovesOn(false);
+    }
+    selectedLayerRef.current = selectedLayer;
+    console.log("called");
```
Comment (gislawill) [inline src/compass/compass.js:78]: A console.log to remove

File: src/compass/compass.js [hunk 10]
```diff
@@ -28,10 +30,18 @@ function Tooltip({ text }) {
...
+function MangroveContextLayer({
+  map,
+  mapLoaded,
+  theme,
+  action,
+  selectedLayer,
+}) {
+  console.log(selectedLayer);
   const [mangrovesOn, setMangrovesOn] = useState(false);
   const [filterIsHovering, setFilterIsHovering] = useState(false);
```
Comment (gislawill) [inline src/compass/compass.js:42]: I don't think this is doing anything, can we remove it?

Comment (gislawill) [reply src/compass/compass.js:42]: "This" meaning the filterIsHovering state

**Pull request closed without merging**